paddle/fluid/operators/fused/mkldnn/fusion_lstm_mkldnn_op.cc

```diff
@@ -0,0 +1,378 @@
+/* Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#include "paddle/fluid/operators/fused/fusion_lstm_op.h"
+#include "paddle/fluid/operators/fused/mkldnn/fusion_rnn_mkldnn.h"
+
+namespace paddle {
+namespace operators {
+
+using paddle::framework::LoDTensor;
+using paddle::framework::Tensor;
+using paddle::platform::CPUDeviceContext;
+using paddle::platform::CreateKey;
+using paddle::platform::MKLDNNGetDataType;
+using paddle::platform::MKLDNNMemDesc;
+using platform::to_void_cast;
+
+template <typename T, typename T_out = T>
+class LSTMMKLDNNHandler
+    : public RNNMKLDNNHandler<T, dnnl::lstm_forward, T_out> {
+ public:
+  LSTMMKLDNNHandler(const paddle::framework::ExecutionContext& ctx,
+                    const platform::MKLDNNDeviceContext& dev_ctx,
+                    const mkldnn::engine mkldnn_engine,
+                    platform::Place cpu_place, const LoDTensor* input,
+                    const Tensor* weight_h, const Tensor* h0, const Tensor* c0,
+                    const bool is_reverse, const int64_t N, const int64_t Ti,
+                    const int64_t IC, const int64_t OC,
+                    const std::string& unique_name)
+      : RNNMKLDNNHandler<T, dnnl::lstm_forward, T_out>(
+            ctx, dev_ctx, mkldnn_engine, ctx.GetPlace(), input, weight_h, h0,
+            is_reverse, N, Ti, IC, OC, 4,
+            ctx.InputName("X") + ctx.InputName("WeightH")) {
+    const bool is_INT8 = std::is_same<T, uint8_t>::value;
```

Comment:
Getting attributes takes some time , so It would be good to move it inside of block (!this->isCached())... to get attributes only once as they are needed only when PD is created.

Reply:
done